amundsen_application/static/js/components/TableDetail/RequestMetadataForm/index.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import './styles.scss';
+
+import { NotificationType, SendNotificationOptions } from 'interfaces/Notifications';
+import { GlobalState } from 'ducks/rootReducer';
+import { connect } from 'react-redux';
+import {
+  TITLE_TEXT,
+  FROM,
+  TO,
+  REQUEST_TYPE,
+  TABLE_DESCRIPTION,
+  COLUMN_DESCRIPTIONS,
+  ADDITIONAL_DETAILS,
+  SEND_BUTTON,
+} from './constants'
+import { SubmitNotificationRequest } from 'ducks/notification/types';
+import { submitNotification } from 'ducks/notification/reducer';
+import { bindActionCreators } from 'redux';
+
+interface StateFromProps {
+  userEmail: string;
+  tableName: string;
+  tableOwners: Array<string>;
+}
+
+export interface DispatchFromProps {
+  submitNotification: (
+    recipients: Array<string>,
+    sender: string,
+    notificationType: NotificationType,
+    options?: SendNotificationOptions
+  ) => SubmitNotificationRequest;
+}
+
+type RequestMetadataProps = StateFromProps & DispatchFromProps;
+
+interface RequestMetadataState {
+  isOpen: boolean,
+}
+
+export class RequestMetadataForm extends React.Component<RequestMetadataProps, RequestMetadataState> {
+  public static defaultProps: Partial<RequestMetadataProps> = {};
```

Comment:
nit: You don't necessarily need this, as `defaultProps` only need to be assigned for optional props. No need to take action, just mentioning for understanding.